### PR TITLE
Drop Python 3.7 tests, add 3.9 tests

### PR DESCRIPTION
**Pull request by dstansby**
This updates the CI to:

- Drop Python 3.7
- Add Python 3.9 (fixes https://github.com/brainglobe/cellfinder-napari/issues/83)
- Since this repository is pure Python, testing on windows/macOS/linux on all python version s (9 jobs) is a bit much, so only run one job each for windows/macos
- Run the linting and manifest jobs in parallel, to reduce the amount of time spent before starting to run the actual tests

There are issues with Python 3.10 tests that I don't understand, so I'm going to leave that for another issue/PR.

### CODE REVIEW

**Review by paddyroddy — APPROVED** [review]
Sounds like a sensible change

**Comment by paddyroddy** [comment]
Although I don't think it's working as you expected
image

**Comment by dstansby** [comment]
If you click into the actual tests, it's a real failure:
```python
   self = <magicgui.backends._qtpy.widgets.Label object at 0x7f023c519990>
  value = 241.375
  
      def _mgui_set_min_width(self, value: int) -> None:
          """Set the minimum allowable width of the widget."""
  >       self._qwidget.setMinimumWidth(value)
  E       TypeError: setMinimumWidth(self, int): argument 1 has unexpected type 'float'
  
  .tox/py310/lib/python3.10/site-packages/magicgui/backends/_qtpy/widgets.py:90: TypeError
```

It's really odd, but shall try and debug/fix

**Comment by dstansby** [comment]
I decided to drop trying to work out Python 3.10 in this PR, so we can get the other changes in.

File: tox.ini [hunk 1]
```diff
@@ -1,24 +1,14 @@
...
-    3.7: py37
     3.8: py38
     3.9: py39
-
-[gh-actions:env]
-PLATFORM =
-    ubuntu-latest: linux
-    macos-latest: macos
-    windows-latest: windows
+    3.10: py310
```
[the remark this answers is not in the document]
Comment (dstansby) [reply tox.ini:9]: 🤷 I can get rid of it if you want, but there's no harm in it being here as far as I can tell.

[the remark this answers is not in the document]
Comment (paddyroddy) [reply tox.ini:9]: I haven't really used `tox` before, and personally I do find it a bit confusing with this still in

**Pull request merged**